### PR TITLE
Ripa questions

### PR DESCRIPTION
**Pull request by Rhian**
proposal_question

### Description of change

We need to find the question(s) that have a flag that matches the flag that is present in the `result` object that RIPA sends us. These are displayed with a number that matches their number in the ordered list in the proposal details array. 

We derive this from the question's index in the proposal_details array, offset by +1.

### Story Link

https://trello.com/c/I99MPhuL/411-as-an-officer-i-want-to-show-the-questions-that-led-to-ripas-decision

### CODE REVIEW

File: app/models/planning_application.rb [hunk 1]
```diff
@@ -216,6 +216,20 @@ def parsed_proposal_details
...
+    parsed_proposal_details.select { |detail| proposal["metadata"].present? }
+  end
+
+  def proposals_with_flags
+    proposals_with_metadata.select { |proposal| proposal["metadata"]["flags"].present? }
+  end
+
+  def flagged_proposal(flag)
+    proposals_with_flags.select do | proposal|
+        proposal["metadata"]["flags"].any? { |element| element == flag }
```
Comment (james) [inline app/models/planning_application.rb]: indentation

File: app/models/planning_application.rb [hunk 2]
```diff
@@ -216,6 +216,20 @@ def parsed_proposal_details
     proposal_details.present? ? JSON.parse(proposal_details) : []
   end
 
+  def proposals_with_metadata
```
Comment (james) [inline app/models/planning_application.rb:219]: I think 'proposals' is the wrong word here really. I think `proposal_details` is probably wrong too, but I think calling each question/answer a proposal is a further step into confusing words. Struggling to think of something else though. `proposal_questions_with_metadata`? `proposal_detail_answers_with_metadata`? What do you think?

Comment (james) [reply app/models/planning_application.rb:219]: Or I guess just `proposal_details_with_metadata`. That's at least consistent.

File: app/models/planning_application.rb (hunk 1 above)
Comment (james) [reply app/models/planning_application.rb]: If `proposal["metadata"]["flags"].include?(flag)` works, then I think that's a bit more readable

File: app/views/shared/_result_information.html.erb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div class="govuk-accordion__section">
+<div class="govuk-accordion__section result_information">
   <div class="govuk-accordion__section-header">
     <h3 class="govuk-accordion__section-heading">
       <button type="button" id="accordion-default-heading-2" aria-controls="accordion-default-heading-3" class="govuk-accordion__section-button" aria-expanded="false">
```
Comment (james) [inline app/views/shared/_result_information.html.erb:4]: I've just noticed that the id and aria-control id are different here. I must have missed it in the last PR 😬

File: app/views/shared/_result_information.html.erb [hunk 4]
```diff
@@ -27,6 +27,21 @@
...
           </p>
         </div>
+        <% if @planning_application.flagged_proposals(@planning_application.result_flag).present? %>
+          <p class="govuk-body">
+            <strong>Details identified <%= @planning_application.api_user.present? ? "by #{@planning_application.api_user.name}" : "" %> as relevant to the result</strong>
+          </p>
+          <% @planning_application.flagged_proposals(@planning_application.result_flag).each do |proposal_detail| %>
+          <p class="govuk-body">
+            <strong><%= proposal_detail["question"] %></strong>
+          </p>
```
Comment (james) [inline app/views/shared/_result_information.html.erb:39]: indentation :-)

File: app/helpers/planning_application_helper.rb [hunk 5]
```diff
@@ -145,4 +145,8 @@ def cancelled_at(planning_application)
       planning_application.returned_at
     end
   end
+
+  def display_number(proposal_details, element)
+    proposal_details.find_index(element) + 1
```
Comment (james) [inline app/helpers/planning_application_helper.rb:150]: Didn't know about `find_index`!

**Review by james — APPROVED** [review]
I've left some tweaks, but no blockers

File: app/views/shared/_result_information.html.erb (hunk 3 above)
Comment (Rhian) [reply app/views/shared/_result_information.html.erb]: Badly copy-pasted from elsewhere and forgot to update, sorry 😬

File: app/models/planning_application.rb (hunk 1 above)
Comment (Rhian) [reply app/models/planning_application.rb]: It worked!

**Pull request merged**